plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiConfig.java

```diff
@@ -0,0 +1,234 @@
+    private boolean sizeBasedSplitWeightsEnabled = true;
+    private DataSize standardSplitWeightSize = DataSize.of(128, MEGABYTE);
+    private double minimumAssignedSplitWeight = 0.05;
+    private int maxSplitsPerSecond = Integer.MAX_VALUE;
+    private int maxOutstandingSplits = 1000;
+
+    public List<String> getColumnsToHide()
+    {
+        return columnsToHide;
+    }
+
+    @Config("hudi.columns-to-hide")
+    public HudiConfig setColumnsToHide(String columnsToHide)
+    {
+        this.columnsToHide = COMMA_SPLITTER.splitToStream(nullToEmpty(columnsToHide))
+                .map(s -> s.toLowerCase(ENGLISH))
+                .collect(toImmutableList());
+        return this;
+    }
+
+    @Config("hudi.metadata-enabled")
+    @ConfigDescription("Fetch the list of file names and sizes from metadata rather than storage")
+    public HudiConfig setMetadataEnabled(boolean metadataEnabled)
+    {
+        this.metadataEnabled = metadataEnabled;
+        return this;
+    }
+
+    public boolean isMetadataEnabled()
+    {
+        return this.metadataEnabled;
+    }
+
+    @Config("hudi.use-metastore-for-partitions")
+    @ConfigDescription("By default, partition info is fetched from the metastore. " +
+            "When this config is disabled, then the partition info is fetched using Hudi's partition extractor and relative partition path.")
+    public HudiConfig setUseMetastoreForPartitions(boolean useMetastoreForPartitions)
+    {
+        this.useMetastoreForPartitions = useMetastoreForPartitions;
+        return this;
+    }
+
+    public boolean getUseMetastoreForPartitions()
+    {
+        return this.useMetastoreForPartitions;
+    }
+
+    @Config("hudi.use-parquet-column-names")
+    @ConfigDescription("Access parquet columns using names from the file. If disabled, then columns are accessed using index."
+            + "Only applicable to parquet file format.")
+    public HudiConfig setUseParquetColumnNames(boolean shouldUseParquetColumnNames)
+    {
+        this.shouldUseParquetColumnNames = shouldUseParquetColumnNames;
+        return this;
+    }
+
+    public boolean getUseParquetColumnNames()
+    {
+        return this.shouldUseParquetColumnNames;
+    }
+
+    @Config("hudi.partition-scanner-parallelism")
+    @ConfigDescription("Number of threads to use for partition scanners")
+    public HudiConfig setPartitionScannerParallelism(int partitionScannerParallelism)
+    {
+        this.partitionScannerParallelism = partitionScannerParallelism;
+        return this;
+    }
+
+    @Min(1)
+    @Max(100)
+    public int getPartitionScannerParallelism()
+    {
+        return this.partitionScannerParallelism;
+    }
+
+    @Config("hudi.split-generator-parallelism")
+    @ConfigDescription("Number of threads to use for split generation")
+    public HudiConfig setSplitGeneratorParallelism(int splitGeneratorParallelism)
+    {
+        this.splitGeneratorParallelism = splitGeneratorParallelism;
+        return this;
```

Comment:
For newbies and also future commiters on `trino-hudi` it would be very beneficial to have a `hudi.rst` documentation page for the new connector.